### PR TITLE
Add org.gyunaev.Birdtray

### PR DESCRIPTION
**Pull request by ThanosApostolou**
I'm not the developer but I have contributed the initial appdata and qmake install scripts to the upstream project (https://github.com/gyunaev/birdtray).

The `thunderbird` script and the patch are needed as a workaround for the birdtray's function to launch thunderbird when birdtray is launched. Birdtray has a configurable option for launching thunderbird but it's kinda limited since it's using the binary's path and cannot work by just changing it to `flatpak-spawn ...` command. If this is accepted I'm willing to work with the upstream project and try to make it more flatpak friendly so that it can be built with just a module with qmake buildsystem (without any other scripts or patches).

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.gyunaev.Birdtray

**Comment by ThanosApostolou** [comment]
bot, build org.gyunaev.Birdtray

File: org.gyunaev.Birdtray.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  <name>Birdtray</name>
+  <summary>System tray new mail notification for Thunderbird</summary>
+  <description>
+    <p>
+      Birdtray is a system tray new mail notification for Thunderbird, which does not require extensions
+    </p>
+  </description>
+  <!-- https://specifications.freedesktop.org/menu-spec/latest/apa.html -->
+  <!-- https://specifications.freedesktop.org/menu-spec/latest/apas02.html -->
+  <categories>
```
Comment (TingPing) [inline org.gyunaev.Birdtray.appdata.xml]: These go in the desktop file not the appdata file.

File: org.gyunaev.Birdtray.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ 
+  <project_group>gyunaev</project_group>
+  <project_license>GPL-3.0+</project_license>
+  <developer_name>gyunaev</developer_name>
+  <screenshots>
+    <screenshot type="default">https://raw.githubusercontent.com/ThanosApostolou/flathub/org.gyunaev.Birdtray/birdtray-settings.png</screenshot>
+  </screenshots>
+  
+ 
+  <content_rating type="oars-1.0">
```
Comment (TingPing) [inline org.gyunaev.Birdtray.appdata.xml]: Use the 1.1 format and you can omit all of these `none` entries.

File: org.gyunaev.Birdtray.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.13",
+    "sdk": "org.kde.Sdk",
+    "command": "birdtray",
+    "finish-args": [
+      "--share=ipc",
+      "--socket=x11",
+      "--socket=wayland",
+      "--device=dri",
+      "--filesystem=~/.thunderbird:ro",
```
Comment (TingPing) [inline org.gyunaev.Birdtray.json]: I believe this just fails if the directory doesn't exist? Can you test that.

File: org.gyunaev.Birdtray.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  <!-- https://specifications.freedesktop.org/menu-spec/latest/apas02.html -->
+  <categories>
+    <category>Calendar</category>
+    <category>Chat</category>
+    <category>Communication</category>
+    <category>ContactManagement</category>
+    <category>Email</category>
+    <category>Feed</category>
+    <category>InstantMessaging</category>
+    <category>IRCClient</category>
```
Comment (TingPing) [inline org.gyunaev.Birdtray.appdata.xml]: This is an irc client?

**Comment by ThanosApostolou** [comment]
bot, build org.gyunaev.Birdtray

File: org.gyunaev.Birdtray.appdata.xml (hunk 4 above)
Comment (ThanosApostolou) [reply org.gyunaev.Birdtray.appdata.xml]: This is just an indicator application which monitors thunderbird accounts for new mails. So, I just copied this section from the org.mozilla.Thunderbird appdata. I don't know why thunderbird is listed as irrclient, maybe it has some extension for this. Should I change it?

File: org.gyunaev.Birdtray.json (hunk 3 above)
Comment (ThanosApostolou) [reply org.gyunaev.Birdtray.json]: Tested it and it works fine even when the directory doesn't exist.

**Comment by barthalion** [comment]
Can you add a paragraph to appdata file saying that by default it expects installing Thunderbird from Flathub?

**Comment by ThanosApostolou** [comment]
bot, build org.gyunaev.Birdtray

**Comment by ThanosApostolou** [comment]
@barthalion is it ok now or should I clarify more the description?

**Comment by barthalion** [comment]
No, it's good, thanks!

/merge

**Pull request closed without merging**